src/cursor.js

```diff
@@ -51,53 +40,42 @@ function fetch(ch, path) {
 }
 
 function update(ch, change) {
-  putOnChan(ch, change);
+  putOnChan(ch, Object.freeze(change));
 }
 
 // refining
 function refine(value, updateCh, oldPath, newPath) {
-  const newPathArr = isArray(newPath) ? newPath : newPath.toString().split('.');
-  const refinedValue = value.getIn(newPathArr);
+  const newPathArr = Array.isArray(newPath) ? newPath : newPath.toString().split('.');
+  const refinedValue = getStateByPath(value, newPathArr);
   const refinedPath = oldPath.concat(newPathArr);
 
   return cursor(refinedValue, refinedPath, updateCh);
 }
 
-// array helpers
-function cMap(value, mapper) {
-  return value.map((v, i) => mapper(cursor.refine(i), i));
-}
-
 const cursor = function (value, path, updateCh) {
-  const imVal = Immutable.fromJS(value);
 
   let o = {
-    deref:    partial(deref, imVal),
-    derefJS:  partial(derefJS, imVal),
+    deref:    partial(deref, value),
     path:     path,
 
     replace:  partial(replace, updateCh, path),
-    remove:   partial(remove, updateCh, path, imVal),
+    remove:   partial(remove, updateCh, path, value),
 
-    refine:   partial(refine, imVal, updateCh, path),
+    refine:   partial(refine, value, updateCh, path),
 
     persist:  partial(persist, updateCh, path),
     fetch:    partial(fetch, updateCh, path),
 
-    hasSameValue:  partial(hasSameValue, imVal)
+    equals:   partial(equals, value)
```

Comment:
Changed `hasSameValue` to `equals` so that we can pass cursors to `ramda.equals` and have it work.